### PR TITLE
[CBRD-24516] multi-aggregate-optimize: replace the information which is printed in "show trace" from aggregate function optimization

### PR DESCRIPTION
**Pull request by beyondykk9**
http://jira.cubrid.org/browse/CBRD-24516

SHOW TRACE is will be shown as like below:
```
Trace Statistics:
 SELECT (time: 3736, fetch: 7664, ioread: 0)
   SCAN (table: dba.mx), (heap time: 3505, fetch: 7664, ioread: 0, readrows: 1000000, rows: 1000000, agl: [idx_mx, idx_min])

```

### CODE REVIEW

**Review by hgryoo — APPROVED** [review]

File: src/query/scan_manager.h [hunk 1]
```diff
@@ -302,6 +302,13 @@ struct scan_pos
   QFILE_TUPLE_POSITION ls_tplpos;	/* List file index scan position */
 };				/* Scan position structure */
 
+typedef struct scan_agl SCAN_AGL;
```
Comment (hyunikn) [inline src/query/scan_manager.h:305]: [Q] Does AGL stands for Aggregation List?

File: src/query/query_executor.c [hunk 2]
```diff
@@ -1803,9 +1803,15 @@ qexec_clear_access_spec_list (THREAD_ENTRY * thread_p, XASL_NODE * xasl_p, ACCES
   for (p = list; p; p = p->next)
     {
       /* aggregate optimize related should be free */
-      if (p->s_id.scan_stats.agg_index_name != NULL)
+      if (p->s_id.scan_stats.agl)
 	{
-	  free (p->s_id.scan_stats.agg_index_name);
+	  SCAN_AGL *agl;
+
+	  for (agl = p->s_id.scan_stats.agl; agl; agl = agl->next)
```
Comment (hyunikn) [inline src/query/query_executor.c]: agl->next : accessing next of the freed agl seems unsafe.

File: src/query/scan_manager.h (hunk 1 above)
Comment (beyondykk9) [reply src/query/scan_manager.h:305]: agl stands for Aggregation Lookup.

File: src/query/query_executor.c (hunk 2 above)
Comment (beyondykk9) [reply src/query/query_executor.c]: Ok, I modified at the next commit.

**Review by hyunikn — APPROVED** [review]

File: src/query/query_executor.c [hunk 3]
```diff
@@ -23879,16 +23890,45 @@ qexec_evaluate_aggregates_optimize (THREAD_ENTRY * thread_p, AGGREGATE_TYPE * ag
...
+		  break;
+		}
+	    }
+
+	  if (agl == NULL)
+	    {
+	      agl = (SCAN_AGL *) malloc (sizeof (SCAN_AGL));
+	      if (agl == NULL)
+		{
+		  return ER_FAILED;
```
Comment (ctshim) [inline src/query/query_executor.c:23921]: free (agg_index_name);
return ER_FAILED;

**Review by ctshim — APPROVED** [review]

**Pull request merged**